Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.SceneManagement;
+using UnityEditor;
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Tools for recording and playing back input animation in the Unity editor.
+    /// </summary>
+    public class InputRecordingWindow : EditorWindow
+    {
+        private InputAnimation animation
+        {
+            get { return PlaybackService?.Animation; }
+            set { if (PlaybackService != null) PlaybackService.Animation = value; }
+        }
+
+        private string loadedFilePath = "";
+
+        private IMixedRealityInputRecordingService recordingService = null;
+        private IMixedRealityInputRecordingService RecordingService => recordingService ?? (recordingService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputRecordingService>());
```

Comment:
Super nit: Some of these lines are super long (consider taming them to be like 100 characters).

Normally I wouldn't have seen this issue but since this PR has over a certain number of commits, Codeflow isn't able to open it up, and github CR stuff has a limit on width.